GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Upgrade/ReplaceObjectUpgrade.cpp

```diff
@@ -83,6 +84,10 @@ void ReplaceObjectUpgrade::upgradeImplementation( )
 		return;
 	}
 
+	Drawable* selectedDrawable = TheInGameUI->getFirstSelectedDrawable();
+	Bool oldObjectSelected = selectedDrawable && selectedDrawable->getID() == me->getDrawable()->getID();
+	Int oldObjectSquadNumber = me->getControllingPlayer()->getSquadNumberForObject(me);
```

Comment:
I suggest put the code starting from `Object *me = getObject();` to ending with `TheGameLogic->destroyObject(me);` into a scope, because the destroyed pointers will be dangling after this.